### DISCUSSION TITLE
How to delete an unacked subject message?

### DISCUSSION THREAD

**Discussion by AhmedAbouelkher** [nats-io/nats-server#3444] [Q&A] [answered]
hi,
I want to delete an unacked messages by id for example from the published to the subject.

For example, when the service user performs a destructive queue action, which will need to cancel the already published message.

**Comment by derekcollison** [comment, marked as answer]
We only do de-dupe via MsgID at this time. Referencing messages is by stream sequence or subjects. Once a sequence has been established you can delete the message directly from the stream.

**Reply by AhmedAbouelkher** [reply]
Where can I get `MsgID` and what is `stream sequence` ?

**Reply by AhmedAbouelkher** [reply]
> Where can I get `MsgID` and what is `stream sequence` ?

Ok, currently I nearly know what is `stream sequence` from [here](https://docs.nats.io/running-a-nats-service/nats_admin/jetstream_admin/consumers#stream-vs-consumer-sequence-numbers)

But how to get it from the go client?

**Reply by derekcollison** [reply]
Its part of the PubAck when you call js.Publish()..

**Reply by derekcollison** [reply]
https://github.com/nats-io/nats.go/blob/main/js.go#L421
https://github.com/nats-io/nats.go/blob/main/js.go#L36

**Reply by AhmedAbouelkher** [reply]
> > Where can I get `MsgID` and what is `stream sequence`?
> 
> Ok, currently I nearly know what is `stream sequence` from [here](https://docs.nats.io/running-a-nats-service/nats_admin/jetstream_admin/consumers#stream-vs-consumer-sequence-numbers)
> 
> But how to get it from the go client?

I could find the message sequence here
```
type PubAck struct {
	Stream    string `json:"stream"`
	Sequence  uint64 `json:"seq"`
	Duplicate bool   `json:"duplicate,omitempty"`
	Domain    string `json:"domain,omitempty"`
}
```
